test/api/unit/services/SiteCreator.test.js

```diff
@@ -306,27 +302,85 @@ describe('SiteCreator', () => {
             }],
           });
           return SiteCreator.createSite({ user, siteParams });
-        }).catch((err) => {
+        })
+        .catch((err) => {
           expect(err.status).to.equal(400);
           expect(err.message).to.equal('A repo with that name already exists.');
           done();
-        }).catch(done);
+        })
+        .catch(done);
       });
 
       it('should reject if the template does not exist', (done) => {
-        const siteParams = {
+        const badSiteParams = {
           owner: crypto.randomBytes(3).toString('hex'),
           repository: crypto.randomBytes(3).toString('hex'),
           template: 'not-a-template',
         };
 
-        factory.user().then(user =>
-          SiteCreator.createSite({ user, siteParams })
-        ).catch((err) => {
+        factory.user()
+        .then((model) => {
+          user = model;
+          return SiteCreator.createSite({ user, siteParams: badSiteParams });
+        }).catch((err) => {
           expect(err.status).to.eq(400);
           expect(err.message).to.eq('No such template: not-a-template');
           done();
-        }).catch(done);
+        })
+        .catch(done);
+      });
+    });
+
+    context('creating a site from an existing site', () => {
+      it('creates a repo and webhook, new site on federalist for specified user', (done) => {
+        let user;
+        let site;
+        let siteParams;
+        let webhookNock;
+
+        factory.user()
+        .then((userModel) => {
+          user = userModel;
+
+          return factory.site({ owner: user.username });
+        })
+        .then((siteModel) => {
+          site = siteModel;
+
+          return { site, user };
+        })
+        .then((values) => {
+          siteParams = {
+            owner: crypto.randomBytes(3).toString('hex'),
+            repository: crypto.randomBytes(3).toString('hex'),
+            defaultBranch: 'master',
+            source: {
+              owner: values.site.owner,
+              repo: values.site.repository,
+            },
+          };
+
+          githubAPINocks.repo();
+          githubAPINocks.createRepoForOrg();
```

Comment:
I'm not super familiar with these nocks -- do they properly test what we need tested without any params specified to them?

Reply:
I think that in this case, we don't really care about the outcome of the repo creation, we just want the test to proceed as though the repo was created. Not passing any args basically just makes the request return `true`